lib/handlers/subscribe.js

```diff
@@ -152,26 +152,24 @@ function completeSubscribe (err) {
   const packet = this.packet
   const client = this.client
   const broker = client.broker
+
   const granted = this.subState.map(obj => obj.granted)
   this.subState = []
-
   const subs = packet.subscriptions
 
-  if (!packet.restore) {
-    broker.emit('subscribe', subs, client)
-  }
-
+  // no messageId conditions:
+  // - restore sub, won't have messageId
+  // - programatically trigger subscribe without messageId
```

Comment:
Where does this `restore` property come from? Maybe we could avoid setting it in the first place.